src/chart/animator/animation.cpp

```diff
@@ -110,23 +110,54 @@ void Animation::addKeyframe(const Gen::PlotPtr &next,
 	}
 
 	auto begin = target;
+
+	std::size_t real_animation{};
+
 	if (intermediate0) {
-		addKeyframe(target,
+		real_animation += strategy == RegroupStrategy::fade
+		               || !begin->getOptions()->looksTheSame(
+		                   *intermediate0->getOptions());
+		begin = intermediate0;
+	}
+
+	if (intermediate1) {
+		real_animation += strategy != RegroupStrategy::aggregate
+		               || !begin->getOptions()->looksTheSame(
+		                   *intermediate1->getOptions());
+		begin = intermediate1;
+	}
+
+	real_animation +=
+	    strategy == RegroupStrategy::fade
+	    || !begin->getOptions()->looksTheSame(*next->getOptions());
+
+	auto real_options = options;
+	if (auto &duration = real_options.all.duration;
+	    real_animation > 1 && duration) {
+		*duration =
+		    ::Anim::Duration{static_cast<double>(duration.value())
+		                     / static_cast<double>(real_animation)};
```

Comment:
We should take all.delay parameter into consideration as well.
Maybe it would be better solution to have an Animation::reTime() function here similar to Planner::reTime() instead of precalculating the durations before setting the keyframes.

Reply:
Should we "scale" the pre-generated keyframes based on inner default animation?

Reply:
I solve it with pre-calculation. Is it good, or should I rewrite it?